java/code/src/com/suse/manager/webui/controllers/ProductsController.java

```diff
@@ -319,6 +318,10 @@ public static String data(Request request, Response response, User user) {
                                             s.getStatus(),
                                             s.getChannels().stream().map(c ->
                                                     new ChannelJson(
+                                                            s.getStatus().equals(MgrSyncStatus.INSTALLED) &&
+                                                                    ChannelFactory.lookupByLabel(c.getLabel()) != null ?
+                                                                    ChannelFactory.lookupByLabel(c.getLabel()).getId() :
+                                                                    -1L,
```

Comment:
I think this is too expensive and there is a better way to get the ID.

Check line 300: there is a map "channelByLabel" of all Channels. Please use that map to get the Channel and the ID.

Reply:
Thanks for the hint. Fixed